curriculum/challenges/english/25-front-end-development/lab-rpg-character/67d83df6f82eda3868dd2a84.md

```diff
@@ -0,0 +1,227 @@
+---
+id: 67d83df6f82eda3868dd2a84
+title: Build an RPG Character
+challengeType: 27
+dashedName: build-an-rpg-character
+---
+
+# --description--
+
+Fulfill the user stories below and get all the tests to pass to complete the lab.
+
+**User Stories:**
+
+1. You should have a function named `create_character`.
+1. The function should accept in order a character name and three stats, strength, intelligence, and charisma.
+1. The character name should be validated:
+   - If the character name is not a string, the function should return `The character name should be a string`.
+   - If the character name is longer than 10 characters, the function should return `The character name is too long`.
+   - If the character name contains spaces, the function should return `The character name should not contain spaces`.
+1. The stats should also be validated:
+   - If one or more stats are not integers, the function should return `All stats should be integers`.
+   - If one or more stats are less than 1, the function should return `All stats should be no less than 1`.
+   - If one or more stats are more than 4, the function should return `All stats should be no more than 4`.
+   - If the sum of all stats is different than 7, the function should return `The character should start with 7 points`.
+1. If all values pass the verification, the function should return a string with four lines:
+   - the first line should contain the character name
+   - lines 2-4 should start with the stat abbreviation, `STR`, `INT` or `CHA` (in this order), then a space, and then a number of full dots (`●`) equal to the value of the stat, and a number of empty dots (`○`) to reach 10. Example: if the value of strength is 3 there must be 3 full dots followed by 7 empty dots. The dots are given in the editor.
+
+Here's the string `create_character("ren", 4, 2, 1) should return:
+
+   ```md
+   ren
+   STR ●●●●○○○○○○
+   INT ●●○○○○○○○○
+   CHA ●○○○○○○○○○
+   ```
+
+NOTE: while `str` and `int` are common abbreviations for the stats, remember that those are reserved keywords in Python and should not be used as variable names.
+
+# --hints--
+
+You should have a function named `create_character`.
+
+```js
+({
+    test: () => runPython(
+`
+assert _Node(_code).has_function('create_character')
+`
+    )
+})
+```
+
+When `create_character` is called with a first argument that is not a string it should return `The character name should be a string`.
+
+```js
+({
+    test: () => runPython(
+        `
+wrong_types = [3, 5, {"a": 3}, 3.1, [1, 2], (1, 2), {1, 2}]
+for wrong_type in wrong_types:
+    assert create_character(wrong_type, 1, 2, 4) == 'The character name should be a string'
+
+# even if wrong numbers, wrong type has precedence
+
+assert create_character(1, 1, 1, 1) == 'The character name should be a string'
+`
+    )
+})
+```
+
+When `create_character` is called with a first argument that contains a space it should return `The character name should not contain spaces`.
```

Comment:
This test and the one right after it should be switched. 

When I was testing this, I accidentally forget the `len()` function here.
However, the first failing test shown has to deal with spaces which can be confusing for campers especially if they did that part right 

<img width="943" alt="Screenshot 2025-04-20 at 1 59 44 PM" src="https://github.com/user-attachments/assets/478e7f5d-f108-4889-a91b-c8ba7c2a20c2" />